src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,145 @@
+import networking.ICommunicator;
+import java.util.HashMap;
+
+/**
+ * This class ContentCommunicator implements the IContentCommunicator interface and all the 
+ * methods inside it. The ideal purpose of this class is to initialise any new client, send 
+ * message from UI to networking module, handle client exit with respect to content module
+ * and handle the subscribers of content module.
+ * 
+ * @author Badal Kumar (111701008)
+ */
 public class ContentCommunicator implements IContentCommunicator{
-    
-	@Override
+	/**
+	 * This variable will store the username/userid of client as a String.
+	 */
+	private String userName;
+	
+	/**
+	 * This variable will store the image of client in String format.
+	 */
+	private String userImage;
+	
+	/**
+	 * This variable will store the server's ipAddress as a String.
+	 */
+	private String serverIPAddress;
+	
+	/**
+	 * This variable will store the port number of the board which the client is accessing.
+	 */
+	private int port = 0;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private HashMap<String, String> mapImage = new HashMap<String, String>();
+	
+	/**
+	 * Creating a HashMap to store the identifier and handler of subscriber of this module,
+	 * identifier is of type String while handler is of type IContentNotificationHandler.
+	 */
+	private HashMap<String, IContentNotificationHandler> mapHandler = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * This variable will store the communicator of networking module to send data over the network.
+	 */
+	private ICommunicator communicator;
+	
+	/**
+	 * This method provides userName to other classes within this package.
+	 * @return userName - username of the client.
+	 */
+	protected String getUserName() {
+		return userName;
+	}
+	
+	/**
+	 * This method provides HashMap mapImage to other classes within this package.
+	 * @return mapImage - The HashMap which stores the userName and userImage.
+	 */
+	protected HashMap<String, String> getMapImage() {
+		return mapImage;
+	}
+	
+	/**
+	 * This method allows other classes within this package to set the HashMap mapImage.
+	 * @param mapImage - The HashMap which stores the userName and userImage of all clients.
+	 */
+	protected void setMapImage(HashMap<String, String> mapImage) {
+		this.mapImage = mapImage;
+	}
+	
+	/**
+	 * This method provides HashMap mapHandler to other classes within this package.
+	 * @return mapHandler - The HashMap which stores the identifier and handler of subscriber of this module.
+	 */
+	protected HashMap<String, IContentNotificationHandler> getMapHandler() {
+		return mapHandler;
+	}
+	
+	/**
+	 * This method initializes a user by storing its username and userimage in a map as Strings.
+	 * This method stores these values locally too and generates server's ipAddress by getting
+	 * the port value of the board from ServerPort class.

```

Comment:
Comma at end

Reply:
If you hover your cursor on the method name in eclipse, you will see a comma in between the line then. In my opinion, comma is nit needed. It will make the description wierd.